lib/cartopy/mpl/geoaxes.py

```diff
@@ -1497,25 +1513,6 @@ def contourf(self, *args, **kwargs):
             A :class:`~cartopy.crs.Projection`.
 
         """
-        t = kwargs.get('transform', None)
-        if t is None:
-            t = self.projection
-        if isinstance(t, ccrs.CRS) and not isinstance(t, ccrs.Projection):
-            raise ValueError('invalid transform:'
-                             ' Spherical contouring is not supported - '
-                             ' consider using PlateCarree/RotatedPole.')
-        if isinstance(t, ccrs.Projection):
-            kwargs['transform'] = t = t._as_mpl_transform(self)
-        else:
-            kwargs['transform'] = t
-
-        # Set flag to indicate correcting orientation of paths if not ccw
-        if isinstance(t, mtransforms.Transform):
```

Comment:
`ccrs.Projection` is not a `mtransforms.Transform`, but the removed `_as_mpl_transform` would have converted it to a `InterProjectionTransform + axes.transData` which _is_. So this code path would not be triggered in the new version.

Strangely, the code that reads this (somewhere else in this file only), uses `getattr` with a default of `False`. I don't know which one is right. Since this is only trying to affect `InterProjectionTransform`, I wonder why that class doesn't just set a default `force_path_ccw` itself.

This is the only one that checks this though, right, so maybe you can just revert this?

Reply:
I agree that it is actually cleaner if I put it back in here since this is the only place it is used, so I removed the force_ccw from the _add_transform decorator and put it back here. I get a bit lost reading through all of the Transform pipelines here, and all of the various branches/possibilities with special casing.

Reply:
Thanks for putting it back, but you missed the first part of my comment, namely that the `_as_mpl_transform` would make `Projection` into a `Transform` that would match this condition.

Reply:
Ahh, I see that now. So, I have now added that back in here as well.

However, another thing to consider is that the `_as_mpl_transform()` was actually used in a lot of the other functions as well. But, seemingly has no effect for those ones and only this code path as you pointed out. I'm not entirely clear whether the call should be put in the decorator or just left here for this one function. Thoughts?

Reply:
Well, tests seem to be okay with it. I think that was needed for old Matplotlib versions which didn't handle it itself. Eventually, the conversion was upstreamed, so this is no longer needed here.